AddressBook.java

```diff
@@ -0,0 +1,1317 @@
+            showToUser(String.format(MESSAGE_INVALID_FILE, filePath));
+            exitProgram();
+        }
+
+        storageFilePath = filePath;
+        createFileIfMissing(filePath);
+    }
+
+    /**
+     * Displays the goodbye message and exits the runtime.
+     */
+    private static void exitProgram() {
+        showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
+        System.exit(0);
+    }
+
+    /**
+     * Sets up the storage based on the default file.
+     * Creates file if missing.
+     * Exits program if the file cannot be created.
+     */
+    private static void setupDefaultFileForStorage() {
+        showToUser(MESSAGE_USING_DEFAULT_FILE);
+        storageFilePath = DEFAULT_STORAGE_FILEPATH;
+        createFileIfMissing(storageFilePath);
+    }
+
+    /**
+     * Returns true if the given file is acceptable.
+     * The file path is acceptable if it ends in '.txt'
+     * TODO: Implement a more rigorous validity checking.
+     */
+    private static boolean isValidFilePath(String filePath) {
+        return filePath.endsWith(".txt");
+    }
+
+    /**
+     * Initialises the in-memory data using the storage file.
+     * Assumption: The file exists.
+     */
+    /*
+    private static void loadDataFromStorage() {
+        initialiseAddressBookModel(loadPersonsFromFile(storageFilePath));
+    }
+    */
+
+
+    /*
+     * ===========================================
+     *           COMMAND LOGIC
+     * ===========================================
+     */
+
+    /**
+     * Executes the command as specified by the {@code userInputString}
+     *
+     * @param userInputString  raw input from user
+     * @return  feedback about how the command was executed
+     */
+    public static String executeCommand(String userInputString) {
+        final String[] commandTypeAndParams = splitCommandWordAndArgs(userInputString);// this method will return an array of Strings in which the first index will be for the argument and the second for the commnand typed by the user. 
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+        switch (commandType) {
+        case COMMAND_ADD_WORD:
+            return executeAddPerson(commandArgs);
+        case COMMAND_FIND_WORD:
+            return executeFindPersons(commandArgs);
+        case COMMAND_LIST_WORD:
+            return executeListAllPersonsInAddressBook();
+        case COMMAND_DELETE_WORD:
+            return executeDeletePerson(commandArgs);
+        case COMMAND_CLEAR_WORD:
+            return executeClearAddressBook();
+        case COMMAND_HELP_WORD:
+            return getUsageInfoForAllCommands();
+        case COMMAND_EXIT_WORD:
+            executeExitProgramRequest();
+        case "sort":
+            executeSortRequest();
+            break;
+        case "edit":
+            executeEditRequest(commandArgs);
+            break;
```

Comment:
would be better if this follows the format for the other commands i.e. `return executeEditRequest(commandArgs)` instead of using `break` so that users can get feedback as well